build/build.go

```diff
@@ -1218,15 +1219,20 @@ func waitContextDeps(ctx context.Context, index int, results *waitmap.Map, so *c
 				}
 				so.FrontendInputs[k+"::"+platform] = st
 				so.FrontendAttrs[v+"::"+platform] = "input:" + k + "::" + platform
-				dt, ok := rr.Metadata["containerimage.config/"+platform]
-				if !ok {
-					continue
+				metadata := make(map[string][]byte)
+				if dt, ok := rr.Metadata[exptypes.ExporterImageConfigKey+"/"+platform]; ok {
+					metadata[exptypes.ExporterImageConfigKey] = dt
 				}
-				dt, err = json.Marshal(map[string][]byte{"containerimage.config": dt})
-				if err != nil {
-					return err
+				if dt, ok := rr.Metadata[exptypes.ExporterBuildInfo+"/"+platform]; ok {
+					metadata[exptypes.ExporterBuildInfo] = dt
+				}
+				if len(metadata) > 0 {
+					dt, err := json.Marshal(metadata)
+					if err != nil {
+						return err
+					}
+					so.FrontendAttrs["input-metadata:"+k+"::"+platform] = string(dt)
```

Comment:
We should think about how to clean up this `k` so it isn't the `resultKey()` that is a bake artifact. If it ends up in buildinfo as "deps" key it should be a clean value.